src/gui/src/abstractpreview.cpp

```diff
@@ -0,0 +1,1575 @@
+        p->setPen(col);
+        p->setBrush(colb);
+        p->drawRect(rr);
+        // p->setCompositionMode(QPainter::CompositionMode_SourceOver);
+      }
+      paintBorder(p);
+
+      //Сохраняем параметры прорисованных страниц
+      QPageInfo pg(page,
+                   QRect(xx - d->ms_left, yy - d->ms_top,
+                         d->paperWidthToScale(), d->paperHeightToScale()));
+      d->pagesRect.append(pg);
+
+      //Дополнительная прорисовка страницы
+      QRect rr(0, 0, viewport()->width() - 1, viewport()->height() - 1);
+      QRect rr1 = pg.rect().intersected(rr);
+      p->save();
+      additionalPaintEvent(p, page, rr1); //Дополнительная прорисовка страниц
+      p->restore();
+
+      p->restore();
+
+      if (d->pageLabelHeight) {
+        p->setPen(palette().color(QPalette::Text));
+        p->drawText(d->pageLabelRect, Qt::AlignCenter,
+                    QString::number(page + 1));
+      }
+
+      page++;
+      if (page == d->pagesData.count())
+        break;
+      xx += (d->paperWidthToScale() + d->interPageSpacing);
+      p->translate((d->paperWidthToScale() + d->interPageSpacing), 0);
+    } // end for(int x = 0; x < d->cols; x++)
+
+    if (d->viewMode != Mode_Thumbs) {
+      int _y = yy - d->ms_top;
+      if (_y < (viewport()->height() >> 1)) {
+        d->selectPageIndex = page - 1;
+      }
+    }
+
+    p->restore();
+    if (page == d->pagesData.count())
+      break;
+    p->translate(0, d->paperHeightToScale() + d->interPageSpacing +
+                        d->pageLabelHeight);
+    yy += (d->paperHeightToScale() + d->interPageSpacing + d->pageLabelHeight);
+  } // end for(int y = 0; y < d->countPagesRow; y++)
+  p->restore();
+  //        for(int i = 0; i < d->pagesRect.count(); i++) {
+  //            p->setPen(Qt::red);
+  //            p->drawRect(d->pagesRect[i].rect());
+  //        }
+  delete p;
+
+  if (d->viewMode != Mode_Thumbs && d->selectPageIndex >= 0) {
+    emit currentPage(d->selectPageIndex);
+  }
+}
+//--------------------------------------------------------------
+void AbstractPreview::additionalPaintEvent(QPainter *p, int page,
+                                           const QRect &rect) {
+  Q_UNUSED(p);
+  Q_UNUSED(page);
+}
+//--------------------------------------------------------------
+QSize AbstractPreview::getPaperSize() { return d->paperSize; }
+QSize AbstractPreview::getPageSize() { return d->pageSize; }
+//--------------------------------------------------------------
+int AbstractPreview::page(QPoint point) {
+  const int count = d->pagesRect.count();
+  int sel = -1;
+  for (int i = 0; i < count; i++) {
+    if (d->pagesRect[i].rect().contains(point)) {
+      sel = d->pagesRect[i].page();
+      break;
+    }
+  }
+
+  if (sel >= 0)
+    return sel;
+  return -1;
+}
+//--------------------------------------------------------------
+QRect AbstractPreview::pageRect(QPoint point) {
+  const int count = d->pagesRect.count();
+  int sel = -1;
```

Comment:
Codacy found an issue: [Variable 'sel' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)